StripePaymentSheet/StripePaymentSheetTests/PaymentSheet/RightAccessoryButtonTest.swift

```diff
@@ -0,0 +1,83 @@
+//
+//  RightAccessoryButtonTest.swift
+//  StripePaymentSheetTests
+//
+//  Created by Nick Porter on 5/31/24.
+//
+
+import Foundation
+@_spi(STP) @testable import StripePaymentSheet
+import XCTest
+
+final class RightAccessoryButtonTest: XCTestCase {
+
+    // MARK: No saved payment methods
+    func testNil_Returned_When_NoSavedPaymentMethods() {
+        // Regardless of any flags, if we have zero payment methods we should return nil
+
+        // Iterating over all combinations of isFirstCardCoBranded, isCBCEligible, allowsRemovalOfLastSavedPaymentMethod, and paymentMethodRemove
+        let booleans = [false, true]
+        for isFirstCardCoBranded in booleans {
+            for isCBCEligible in booleans {
+                for allowsRemovalOfLastSavedPaymentMethod in booleans {
+                    for paymentMethodRemove in booleans {
+                        let result = RowButton.RightAccessoryButton.getAccessoryButtonType(savedPaymentMethodsCount: 0,
+                                                                                           isFirstCardCoBranded: isFirstCardCoBranded,
+                                                                                           isCBCEligible: isCBCEligible,
+                                                                                           allowsRemovalOfLastSavedPaymentMethod: allowsRemovalOfLastSavedPaymentMethod,
+                                                                                           paymentMethodRemove: paymentMethodRemove)
+                        XCTAssertNil(result)
+                    }
+                }
+            }
+        }
+    }
+
+    // MARK: 1 card
+
+    func testSingleCard_Returns_CorrectValue() {
+        // Iterating over all combinations of isFirstCardCoBranded, isCBCEligible, allowsRemovalOfLastSavedPaymentMethod, and paymentMethodRemove
+        let booleans = [false, true]
+        for isFirstCardCoBranded in booleans {
+            for isCBCEligible in booleans {
+                for allowsRemovalOfLastSavedPaymentMethod in booleans {
+                    for paymentMethodRemove in booleans {
+
+                        // We should only show the edit button if it is update-able or removable
+                        let canEdit = (isFirstCardCoBranded && isCBCEligible) || (allowsRemovalOfLastSavedPaymentMethod && paymentMethodRemove)
```

Comment:
Don't love this logic living in the test, but it's a tradeoff to get a very exhaustive set of tests on something that is edge-casey.

Reply:
Yeah I'm not sure how I feel about these tests, hah. It feels so tightly coupled to the implementation of the function it's testing. I think I'd prefer if we tested this at a higher level so it can detect bugs outside of this simple method. I dont' feel strongly right now, can follow up.